packages/ui/src/components/DataStory/clients/JsClient.tsx

```diff
@@ -54,12 +70,24 @@ export class JsClient implements ServerClient {
       type: DataStoryEvents.RUN_START
     });
 
-    const storage = new InMemoryStorage()
+    const storage = new InMemoryStorage();
+    const sendMsg: NotifyObserversCallback = ( InputObserver: InputObserver, items: ItemValue[]) => {
+      this.reportLinkItems?.watchDataChange(
+        InputObserver,
+        items
+      )
+    }
+
+    const inputObserverController = new InputObserverController(
+      this.reportLinkItems?.inputObservers || [],
+      sendMsg
+    );
 
-    this.executor = ExecutorFactory.create(
```

Comment:
In general, executors should now be created using `ExecutorFactoryfactory.create`.
That is because the factory will take care of any preparations needed, such as unfolding any subdiagrams.
That will let Executor have a constructor without any operations.

Reply:
I'm sorry, this issue was introduced while I was resolving conflicts across different branches :sweat_smile: